src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift

```diff
@@ -1679,34 +1735,53 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
 				}
 			)
 		}
-
-		model.registerTestAndGetResult(
-			for: testQRCodeInformation,
-			isSubmissionConsentGiven: submissionConsentGiven,
-			certificateConsent: certificateConsent,
-			isLoading: isLoading,
-			onSuccess: { [weak self] testResult in
-				
-				self?.model.coronaTestType = testQRCodeInformation.testType
-
-				switch testQRCodeInformation {
-				case .teleTAN:
-					self?.showTestResultScreen()
-				case .antigen, .pcr, .rapidPCR:
-					switch testResult {
-					case .positive:
-						self?.showTestResultAvailableScreen()
-					case .pending, .negative, .invalid, .expired:
+		
+		switch testOwner {
+		case .user:
+			guard let isSubmissionConsentGiven = submissionConsentGiven else { return }
```

Comment:
Do I understand it correctly. If the submission consent is nil, then there won't be a test result?

Reply:
In case of .user the submissionConsentGiven should never be nil, either it is true or false. This statement make sure this.